test/test_external.py

```diff
@@ -298,6 +299,29 @@ def test_private_space(self):
         except TooManyRequestsError:
             pass
 
+    def test_multiple_spaces_one_private(self):
+        api_key = "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"  # Intentionally revealing this key for testing purposes
+        with gr.Blocks():
+            gr.Interface.load(
+                "spaces/gradio-tests/not-actually-private-space", api_key=api_key
+            )
+            gr.Interface.load(
+                "spaces/gradio/test-loading-examples",
+            )
+        assert Context.access_token == api_key
+
+    def test_loading_files_via_proxy_works(self):
+        api_key = "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"  # Intentionally revealing this key for testing purposes
+        io = gr.Interface.load(
+            "spaces/gradio-tests/test-loading-examples-private", api_key=api_key
+        )
+        app, _, _ = io.launch(prevent_thread_lock=True)
+        test_client = TestClient(app)
+        r = test_client.get(
+            "/proxy=https://gradio-tests-test-loading-examples-private.hf.space/file=/tmp/tmprahzj703/Bunnyual53t2x.obj"
+        )
+        assert r.status_code == 200
+
 
 class TestLoadInterfaceWithExamples:
     def test_interface_load_examples(self, tmp_path):
```

Comment:
I don't know where else to put this comment lol but we should get rid of `tmp.zip` from version control, no?

Reply:
Haha yes let me do that